android/src/app/organicmaps/location/LocationHelper.java

```diff
@@ -369,6 +371,41 @@ public void stop()
     mActive = false;
   }
 
+  /**
+   * Resume location services when entering the foreground.
+   */
+  public void resumeLocationInForeground()
```

Comment:
Can a user turn-off locations or permissions while in background?
Is it real difference between autostartLocation vs resumeLocationInForeground?

Reply:
> Can a user turn-off locations or permissions while in background?

Android just kills the app in this case. 

> Is it real difference between autostartLocation vs resumeLocationInForeground?

Yes, autostartLocation() request permissions... I can't request permissions properly without Activity.